ecosystem/platform/server/app/controllers/welcome_controller.rb

```diff
@@ -6,14 +6,48 @@
 class WelcomeController < ApplicationController
   layout 'it1'
 
-  before_action :ensure_discord!, only: %i[it1]
   before_action :ensure_confirmed!, only: %i[it1]
 
   def index
-    redirect_to it1_path if current_user && current_user.authorizations.where(provider: :discord).exists?
+    @login_dialog = DialogComponent.new
   end
 
   def it1
     redirect_to root_path unless user_signed_in?
+    @steps = [
+      connect_discord_step,
+      node_registration_step,
+      identity_verification_step
+    ].map { |h| OpenStruct.new(**h) }
+    first_incomplete = @steps.index { |step| !step.completed }
+    @steps[first_incomplete + 1..].each { |step| step.disabled = true } if first_incomplete
+  end
+
+  private
+
+  def connect_discord_step
+    {
+      name: :connect_discord,
+      completed: current_user.authorizations.where(provider: :discord).exists?,
+      dialog: DialogComponent.new
+    }
+  end
+
+  def node_registration_step
+    completed = !current_user&.it1_profile&.validator_verified?.nil?
+    {
+      name: :node_registration,
+      completed:,
+      href: completed ? edit_it1_profile_path(current_user.it1_profile) : new_it1_profile_path
+    }
+  end
+
+  def identity_verification_step
+    completed = !current_user&.kyc_complete?.nil?
```

Comment:
same here with `.nil?`

Reply:
Seems it was a Rubocop "fix". Will revert to what I had before that.